theme/static/js/hs_file_browser.js

```diff
@@ -2745,6 +2745,21 @@ $(document).ready(function () {
         });
     });
 
+    // Unzip to folder method
+    $("#btn-unzip-to-folder, #fb-unzip-to-folder").click(function () {
+        var files = $("#fb-files-container li.ui-selected");
+        var calls = [];
+        for (let i = 0; i < files.length; i++) {
+            let fileName = $(files[i]).children(".fb-file-name").text();
+            calls.push(unzip_irods_file_ajax_submit(SHORT_ID, getCurrentPath().path.concat(fileName).join('/'), overwrite='false', unzip_to_folder='true'));
+        }
+
+        // Wait for the asynchronous calls to finish to get new folder structure
+        $.when.apply($, calls).fail(function () {
```

Comment:
@devincowan Because the ajax call is to trigger the unzip async task, we only want to refresh file browser when the ajax call fails. If it succeeds, we don't want to refresh the file browser while the async unzip task is ongoing. When the async unzip task is done, the user will get the notification from the notification popup asking users to click a link to refresh the page. Does it make sense? I can update the comment to make it clear if this makes sense to you.

Reply:
got it, thanks